packages/contract/src/contract/helpers.ts

```diff
@@ -161,9 +161,10 @@ export function getOptions(
     value?: number | BN,
     gasLimit?: WeightV2,
     storageDeposit?: StorageDeposit,
-    increaseGas?: boolean
+    increaseGas?: boolean,
+    gasIncreaseFactor?: number
 ): ContractOptions {
-    const gasIncreaseFactor = increaseGas ? GAS_INCREASE_FACTOR : 1
+    gasIncreaseFactor = increaseGas ? (gasIncreaseFactor ? gasIncreaseFactor : GAS_INCREASE_FACTOR) : 1
```

Comment:
`gasIncreaseFactor = increaseGas ? gasIncreaseFactor || GAS_INCREASE_FACTOR : 1`

Reply:
You can suggest this as a commit and get it resolved within github. Not sure how you do it though.